Cython/Utility/ModuleSetupCode.c

```diff
@@ -1074,7 +1074,10 @@ static CYTHON_INLINE PyObject * __Pyx_PyDict_GetItemStrWithError(PyObject *dict,
   #define __Pyx_SET_SIZE(obj, size) Py_SIZE(obj) = (size)
 #endif
 
-#if CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS && PY_VERSION_HEX >= 0x030d00b1
+#if (CYTHON_COMPILING_IN_LIMITED_API || CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS) && __PYX_LIMITED_VERSION_HEX >= 0x030d0000
+#define __Pyx_PyList_GetItemRef(o, i) PyList_GetItemRef(o, i)
+#define __Pyx_PyDict_GetItemRef(dict, key, result) PyDict_GetItemRef(dict, key, result)
+#elif CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS && PY_VERSION_HEX >= 0x030d00b1
```

Comment:
Isn't this just covered by the case above? (`__PYX_LIMITED_VERSION_HEX` is just `PY_VERSION_HEX` when limited API is off)